geode-membership/src/main/java/org/apache/geode/distributed/internal/membership/gms/GMSMembership.java

```diff
@@ -1524,15 +1487,9 @@ public boolean isShunned(ID m) {
   }
 
   private boolean isShunnedOrNew(final ID m) {
-    latestViewReadLock.lock();
-    try {
-      return shunnedMembers.containsKey(m) || isNew(m);
-    } finally { // synchronized
-      latestViewReadLock.unlock();
-    }
+    return shunnedMembers.containsKey(m) || isNew(m);
   }
 
-  // must be invoked under view read or write lock
   private boolean isNew(final ID m) {
     return !latestView.contains(m) && !surpriseMembers.containsKey(m);
```

Comment:
I think the read-lock is still needed here.  While the "shunned" state of the member is disconnected from the other two checks, surpriseMembers is modified while forming a new membership view and we want these two checks to be atomic.